core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactory.java

```diff
@@ -308,6 +308,29 @@ RelDataType createDecimalQuotient(
   @SuppressWarnings("deprecation")
   FieldInfoBuilder builder();
 
+  /**
+   * Infers the return type of a decimal addition. Decimal addition involves
+   * at least one decimal operand and requires both operands to have exact
+   * numeric types.
+   *
+   * @param type1 type of the first operand
+   * @param type2 type of the second operand
+   * @return the result type for a decimal addition, or null if decimal
+   * addition should not be applied to the operands.
+   */
+  RelDataType createDecimalAddition(RelDataType type1, RelDataType type2);
+
+  /**
+   * Infers the return type of a decimal mod operation. Currently only a hook point
```

Comment:
The comment about hook point is unnecessary since this is an interface and there's no implementation. Similar to `createDecimalAddition`, should one of the operand be a decimal and the second one be an exact numeric type?

One could (should?) add a default implementation too: the expectation is that it should return something non-null if operands are valid as it might be used in a different context.

Reply:
I've fixed the comment but i do not want to provide a default implementation since that would be change in behavior for all existing clients of calcite.
the change i made preserves the existing behavior.

Reply:
The default should have the same behavior as currently: if one of the operand types are decimal and both are exact numeric types, return the type of the second argument. It doesn't change existing behavior of the operand, and people who use {{RelDataTypeFactory}} can leverage the new method.